### DISCUSSION TITLE
How does ONDC work?

### DISCUSSION THREAD

**Discussion by chainhead** [Open-network-for-digital-commerce/ONDC-Protocol-Specs#62] [General]
# Introduction

The following is a list of questions based upon initial study of the [ONDC strategy paper](https://ondc.org/image/ONDCStrategyPaper.pdf) and [Beckn protocol](https://github.com/beckn).

## Questions

1. The [Unified Payments Interface (UPI)](https://en.wikipedia.org/wiki/Unified_Payments_Interface) is owned and developed by [National Payments Corporation of India](https://en.wikipedia.org/wiki/National_Payments_Corporation_of_India). Similarly, who is expected to be the developer and operator of ONDC?

2. While the goal of connecting the smallest shop to a consumer digitally is noble, the role of curation and governance is paramount for both parties - seller and buyer - to participate in a market place. 
  a. Does ONDC or Beckn have curation and governance baked into the specifications? See 1 above.
  b. Will ONDC play the role of curator and governance enforcement? See 1 above.

3. Given that the Beckn protocol specifications are open, technically it is possible to have multiple "private" implementations of ONDC e.g. all sellers of construction material in Bengaluru, mathematics tutors of a given pin code, etc. Are such multiple private networks envisaged by ONDC? 
  a. If yes, how is curation and governance expected to be enforced? See 2 above.
  b. How are multiple private networks expected to inter-operate?
  c. Buyers and sellers would "multi-home" in multiple, private ONDC networks. How will this be different from the current scenario (e.g. a cab driver present on multiple ride hailing applications)?

4. How are large, incumbent marketplace players classified in ONDC - buyer, seller or both?
  a. How are marketplace players incentivized to on-board to ONDC? See 3 above.
  b. How is a seller on a given marketplace incentivized to on-board to ONDC? See 3 above.
  c. What incentives will ensure ONDC won't be a "super app" or a wrapper on top of large, incumbent marketplace players?

**Comment by ravi-prakash-v** [comment]
Excellent questions @chainhead. Please see my responses inline.

### The [Unified Payments Interface (UPI)](https://en.wikipedia.org/wiki/Unified_Payments_Interface) is owned and developed by [National Payments Corporation of India](https://en.wikipedia.org/wiki/National_Payments_Corporation_of_India). Similarly, who is expected to be the developer and operator of ONDC?

Firstly, as the name suggests, ONDC is a “Network”. It is “developed” by many parties implementing the ONDC Protocol Specification. These developers could be consumer-facing platforms (a.k.a Buyer Apps), business-facing platforms (a.k.a Seller Apps), System Integrators offering tech solutions, Gateways, and Registries. So strictly speaking, no “one” person or organization is considered to be the “developer” of ONDC. 

Regarding who “operates” ONDC, again, ONDC does not have an “operator”. It only has a network “facilitator” which is ONDC Limited. Bear in mind, ONDC and ONDC Limited are two separate entities. You can consider ONDC and ONDC Limited to be analogous to Apache and Apache Software Foundation. While Apache Server is maintained by a community of independent developers, Apache Foundation is an American nonprofit corporation to support a number of open source software projects.  However, ONDC Limited facilitates discussions between various companies big and small to collaborate and expand ONDC (the network). It also advocates ONDC protocol (the adaptation of beckn protocol for retail and logistics) across the globe to expand the network. Sometimes it also hosts some critical network infrastructures like the ONDC Registry, drafts and maintains network policies that are mutually agreed by the network’s participants. The most important thing to note is that ONDC Limited is neither an operator nor a regulator. It is merely a “facilitator”. 

The correct technical term for this is, _Network Facilitator Organization_ or NFO. 

> ONDC Limited is an NFO.

**Reply by chainhead** [reply]
@core-wg-admin Thanks for the detailed responses!

1. I wasn't aware of ONDC Limited. A cursory Google search led me to a [Facebook page](https://www.facebook.com/ONDC-Ltd-103148322189702/?_rdr). I am keen to know if there is a "charter" page for ONDC Limited similar to one from [ASF](https://www.apache.org/foundation/).
2. Who develops the "critical network infrastructures like the ONDC Registry, drafts" that maybe hosted by ONDC Limited?
3. Are network policies machine readable e.g. [XACML](https://www.oasis-open.org/committees/tc_home.php?wg_abbrev=xacml)?

**Comment by ravi-prakash-v** [comment]
### While the goal of connecting the smallest shop to a consumer digitally is noble, the role of curation and governance is paramount for both parties - seller and buyer - to participate in a market place.
### a. Does ONDC or Beckn have curation and governance baked into the specifications? See 1 above.
### b. Will ONDC play the role of curator and governance enforcement? See 1 above.

Regarding curation, ONDC Limited does not “curate” anything. It merely provides the necessary digital infrastructure (like rating and reputation ledgers, Online dispute resolution infrastructure) and policy frameworks to allow curation of buyers and sellers at the “endpoints” of the network. The choice whether to choose a seller to do business with or not is the decision of a buyer and vice-versa. Not ONDC Limited.

Regarding governance, ONDC’s governance model is similar to beckn protocol governance model where there are working groups to address various issues in various categories. ONDC Limited holds weekly community calls, bi-weekly architecture working group calls and monthly technology strategy calls to address important issues and figure out ways to “facilitate” discussions wherever appropriate to address those issues.

**Reply by chainhead** [reply]
> It merely provides the necessary digital infrastructure (like rating and reputation ledgers, Online dispute resolution infrastructure) 
1. Who are authorized for CRUD operations on digital infrastructure?

> policy frameworks to allow curation of buyers and sellers at the “endpoints” of the network.
2. I didn't follow this. Perhaps, you could use this recent announcement about [Lucknow going live on ONDC](https://twitter.com/ONDC_Official/status/1542381227730010112?t=0vEoAoJd1T60vgYob8I2NQ&s=08) to map theory and practice. 

> Regarding governance, ONDC’s governance model is similar to beckn protocol governance model where there are working groups to address various issues in various categories
3. This sounds like setting up guard rails or a _post facto_ approach than something that takes effect at "run time". For example, who/how is a buyer verified before coming on-board? What if a seller can't/won't pay of a Cash on Delivery (CoD) kind of transaction? Suppose, the delivery of product fails (lost, stolen, damaged, etc.) - how does the digital infra-structure respond to such scenarios?

> ONDC Limited holds weekly community calls....
4. Anyway I could get invited?

**Comment by ravi-prakash-v** [comment]
### Given that the Beckn protocol specifications are open, technically it is possible to have multiple "private" implementations of ONDC e.g. all sellers of construction material in Bengaluru, mathematics tutors of a given pin code, etc. Are such multiple private networks envisaged by ONDC?
### a. If yes, how is curation and governance expected to be enforced? See 2 above.
### b. How are multiple private networks expected to inter-operate?
### c. Buyers and sellers would "multi-home" in multiple, private ONDC networks. How will this be different from the current scenario (e.g. a cab driver present on multiple ride hailing applications)?

ONDC as the name suggests, is an **Open** network. By definition it does not envisage or facilitate _private_ networks. However, it does not stop anyone from setting up their private network. The question is if you are using ONDC protocol specification to set up a private network, are you really an “Open Network”? I would rather ask myself, _what do I stand to gain from setting up a private ONDC-like network?_. If the answer makes sense, nothing is stopping you from setting up your own network like ONDC. However, if you are setting up your own private network, you cannot _publicly_ call yourself ONDC. Secondly, if you are using ONDC protocol specification to set up a _new_ ONDC-like network, you must give due attribution to ONDC and beckn protocol specification as per the CC-BY-NC-SA licensing terms of beckn protocol and ONDC protocol. 

Ensuring interoperability between private networks is left to whoever is facilitating their creation. Not ONDC Limited. 

That being said, it is perfectly possible for anyone to be present in more than one network. Private or public, it doesn’t matter. But the policies for two different networks may be different. As long as the participant adheres to both, it should really not matter.

**Reply by chainhead** [reply]
> Ensuring interoperability between private networks is left to whoever is facilitating their creation. Not ONDC Limited. 

Ok. What if a private network wanted to integrate with the ONDC "main" network?

**Comment by ravi-prakash-v** [comment]
### How are large, incumbent marketplace players classified in ONDC - buyer, seller or both?
### a. How are marketplace players incentivized to on-board to ONDC? See 3 above.
### b. How is a seller on a given marketplace incentivized to on-board to ONDC? See 3 above.
### c. What incentives will ensure ONDC won't be a "super app" or a wrapper on top of large, incumbent marketplace players?

Buyers and Sellers, or more specifically Buyer Apps and Seller Apps are “roles”. It is possible for a large player to have both a buyer app and a seller app role. The player playing the role of a Buyer app has access to multiple sellers other than the ones they have acquired themselves. And the same player acting as a seller app has access to  customers other than the ones they have acquired on their own.

Regarding 4a) and 4b). Given ONDC potentially exposes a much larger market to its participants, it is quite natural for any business to “want” more access to the larger market in terms of both “demand” and “supply”. Simply put, a small business owner does not have to spend thousands of marketing dollars to acquire customers. By just being present on ONDC, it automatically has access to the “entire” customer base of the buyer applications out there. Similarly a pure play experience provider does not need to “acquire” businesses to make those products and services accessible to their users. It just has to “plug in” to ONDC and automatically start generating orders. 

Regarding 4c) ONDC is not an app, not a super app, nor a platform. So the question seems largely irrelevant as ONDC is a network consisting of many nodes. These nodes are what form ONDC. So it is highly unlikely that ONDC will ever become a “super-app”. But being part of ONDC gives small businesses and big businesses an equal opportunity to compete by reducing the friction involved in business or consumer acquisition. The rest is left to the platforms to innovate and build their business. ONDC does not limit anyone from growing their business. Whether they are big or small - it is agnostic to ONDC.

**Reply by chainhead** [reply]
> So the question seems largely irrelevant as ONDC is a network consisting of many nodes.

What if, these nodes end-up being almost exclusively owned/operated by large, incumbent marketplace players. Think, 51% attack on a Bitcoin network. Not saying that ONDC is (or should be) a Blockchain, but some of the points called out in this [Pentagon sponsored research](https://www.theregister.com/2022/06/22/research_challenges_assumptions_about_distributed/) describe how Blockchain is not as decentralized as originally conceived.

**Comment by rahulbansal16** [comment]
@chainhead my understanding is that it does not matter if all the nodes are privately owned as long as they adhere to ONDC or bekcn protocol things will work smoothly.
It's like in smtp gmail owns large infra but still anyone can operate and send email to each other.

**Reply by chainhead** [reply]
I am saying, the ONDC may end up as inter operable, multiple private networks via common protocol. I don't know if this is the ONDC vision.
